src/main/java/com/uber/cadence/worker/TimeFilter.java

```diff
@@ -0,0 +1,87 @@
+/*
+ *  Modifications Copyright (c) 2017-2020 Uber Technologies Inc.
+ *  Copyright 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ *  use this file except in compliance with the License. A copy of the License is
+ *  located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed on
+ *  an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+package com.uber.cadence.worker;
+
+import java.time.ZonedDateTime;
+import java.util.Objects;
+
+public class TimeFilter {
+
+    public static TimeFilter.Builder newBuilder() {
+        return new TimeFilter.Builder();
+    }
+
+    public static TimeFilter.Builder newBuilder(TimeFilter options) {
+        return new TimeFilter.Builder(options);
+    }
+
+    public static TimeFilter defaultInstance() {
+        return DEFAULT_INSTANCE;
+    }
+
+    private static final TimeFilter DEFAULT_INSTANCE;
+
+    static {
+        DEFAULT_INSTANCE = TimeFilter.newBuilder().build();
+    }
+
+    public static final class Builder {
```

Comment:
nit: I am not sure if having a Builder for every option field is an overkill or not. But I am not familiar with java client, so feel free to ignore this comment.

Reply:
Had the same feeling. But I incline to keep the consistency here.